### PR TITLE
analytics: add DVC_ANALYTICS env var

### PR DESCRIPTION
**Pull request by efiop**
New special env var to force turn on/off analytics. If not defined - keeps old behavior. Needed for our vs-code plugin to force-disable analytics for them (we don't really care about that analytics and vs-code has its own analytics anyway).

Docs: https://github.com/iterative/dvc.org/pull/2724

### CODE REVIEW

File: dvc/env.py [hunk 1]
```diff
@@ -9,3 +9,4 @@
 DVC_IGNORE_ISATTY = "DVC_IGNORE_ISATTY"
 DVC_EXP_GIT_REMOTE = "DVC_EXP_GIT_REMOTE"
 DVC_EXP_AUTO_PUSH = "DVC_EXP_AUTO_PUSH"
+DVC_ANALYTICS = "DVC_ANALYTICS"
```
Comment (skshetry) [inline dvc/env.py:12]: This should be called `DVC_NO_ANALYTICS` as it's sole purpose is to disable analytics.

See also: https://keepsake.ai/docs/learn/analytics

Comment (efiop) [reply dvc/env.py:12]: @skshetry It is also meant to force-enable analytics, so `NO_ANALYTICS` would get confusing.

Comment (skshetry) [reply dvc/env.py:12]: Why does the user need to force-enable it? Feels unnecessary.

Comment (efiop) [reply dvc/env.py:12]: For the record: continuing in https://github.com/iterative/dvc.org/pull/2724

**Pull request merged**